inst/stan/functions/convolve.stan

```diff
@@ -1,36 +1,100 @@
-// convolve two vectors as a backwards dot product
-// y vector should be reversed
-// limited to the length of x and backwards looking for x indexes
+/**
+ * Calculate convolution indices for the case where s <= xlen
+ *
+ * @param s Current position in the output vector
+ * @param xlen Length of the x vector
+ * @param ylen Length of the y vector
+ * @return An array of integers: {start_x, end_x, start_y, end_y}
+ */
+array[] int calc_conv_indices_xlen(int s, int xlen, int ylen) {
+  int s_minus_ylen = s - ylen;
+  int start_x = max(1, s_minus_ylen + 1);
+  int end_x = s;
+  int start_y = max(1, 1 - s_minus_ylen);
+  int end_y = ylen;
```

Comment:
Can we combine the two functions using an `ifelse` control?

```stan
 if (s <= xlen) {
    end_x = s;
    end_y = ylen;
  } else {
    end_x = xlen;
    end_y = ylen + xlen - s;
  }
```

Reply:
we could but I'm not sure it really simplifies things (your point below)

Reply:
Yep. Was a first thought before seeing the logic below.